File: TwoSum.java
import java.util.HashMap;

public class TwoSum {
    public static void main(String[] args) {
        int[] nums = {11, 15, 3, 7};
        int target = 17;
        System.out.println(two_sum(nums, target));
    }
    public static boolean two_sum(int[] nums, int target) {
        int len = nums.length;
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i=0; i<len; i++) {
            int complement = target - nums[i];
            if(map.containsKey(complement)) {
                return true;
            } else {
                map.put(nums[i], nums[i]);
            }
        }
        return false;
    }
}

File: TwoSum.py
"""
def twoSum(nums, target):
    seen = {}        
    for index, num in enumerate(nums):
        com = target - num            
        if com in seen:
            return True
        else:
            seen[num] = index
    return False;

print(twoSum([1, 2, 3, 4, 5], 3)); """



def twoSum(nums, k):
    map = {};
    for index, num in enumerate(nums):
        complement = k - num;
        if complement in map:
            return True
        else:
            map[num] = num
    return False

print(twoSum([1, 2, 3, 4, 5], 10))